helix-core/src/syntax.rs

```diff
@@ -117,6 +148,25 @@ pub struct IndentQuery {
     pub outdent: HashSet<String>,
 }
 
+#[derive(Debug, Serialize, Deserialize)]
+pub struct GrammarConfiguration {
+    #[serde(rename = "name")]
+    pub grammar_id: String, // c-sharp, rust
+    pub source: GrammarSource,
+    pub path: Option<String>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "lowercase")]
+pub enum GrammarSource {
+    #[serde(rename = "path")]
+    Local(String),
+    Git {
+        remote: String,
+        rev: String,
+    },
+}
```

Comment:
This correctly deserializes the new syntax in `languages.toml` when using `toml::from_slice`, but it now `toml::Value::try_into` reliably fails to convert it (which appears to be expected behavior from [the docs](https://github.com/alexcrichton/toml-rs/blob/499e8c4bb3b75ae6c4dcc91a8bdd81947c577147/src/value.rs#L62-L68)). I think we have to hack around it with a struct that accepts both git remote and path :thinking:

Reply:
Ah that's perfect, thanks!